### PR TITLE
Add hedera app submodules to prep for refactoring

### LINKED ISSUE #3894: Create `hedera-app` and `hedera-app-api` modules under `hedera-node`
These need to be Gradle subprojects and JPMS modules. You can see some examples [here](https://github.com/rbair23/services-poc/tree/main/modules). The POC used Helidon rather than Netty directly, but you really should use netty. Don't bother bringing over the code, just getting the basic build setup and a more or less empty module definition is sufficient for this ticket.

### PR DESCRIPTION
**Pull request by mhess-swl**
**Description**:
This PR creates two new subprojects – `hedera-app-api` and `hedera-app` – under the `hedera-node` project in preparation for a large refactoring effort. See more details in #3893
* Subprojects created
* JPMS module files created

**Related issue(s)**:
Closes #3894

### CODE REVIEW

File: hedera-node/build.gradle.kts [hunk 1]
```diff
@@ -20,6 +20,11 @@ plugins {
 
 description = "Hedera Services Node"
 
+subprojects {
+    project(":hedera-node:hedera-app")
+    project(":hedera-node:hedera-app-api")
+}
+
```
Comment (rbair23) [inline hedera-node/build.gradle.kts:27]: I don't think you need to put this here, I think the lines in settings.gradle.kts are enough?

Comment (mhess-swl) [reply hedera-node/build.gradle.kts]: Gradle is so smart. Fixed

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM!

**Review by kimbor — DISMISSED** [review]

File: hedera-node/hedera-app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ */
+plugins {
+    id("com.hedera.hashgraph.hedera-conventions")
+}
+
+description = "Hedera Application - Implementation"
+
+dependencies {
+    project(":hedera-node:hedera-app-api")
```
Comment (nathanklick) [inline hedera-node/hedera-app/build.gradle.kts:23]: The original code creates an intra-project dependency which spans all source sets with compile, runtime, and transitive dependency scopes. We should be more explicit as to which source sets and what scopes to which this dependency applies.

```suggestion
    api(project(":hedera-node:hedera-app-api"))
```

File: hedera-node/hedera-app/src/main/module-info.java [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module com.hedera.hashgraph.app {
+
```
Comment (nathanklick) [inline hedera-node/hedera-app/src/main/module-info.java:2]: If we mandate this library to depend on the `com.hedera.hashgraph.app.api` module at the Gradle level as compile & runtime scoped dependency, then we should mirror the requirement at the JPMS module level. 

```suggestion
requires com.hedera.hashgraph.app.api;
```

File: hedera-node/hedera-app-api/src/main/module-info.java [hunk 4]
```diff
@@ -0,0 +1,3 @@
+module com.hedera.hashgraph.app.api {
```
Comment (nathanklick) [inline hedera-node/hedera-app-api/src/main/module-info.java:1]: These files must be moved under `src/main/java` as opposed to the current location of `src/main` which places these files outside of all source sets.

**Review by nathanklick — CHANGES_REQUESTED** [review]

File: hedera-node/hedera-app/src/main/module-info.java [hunk 5]
```diff
@@ -0,0 +1,3 @@
+module com.hedera.hashgraph.app {
```
Comment (nathanklick) [inline hedera-node/hedera-app/src/main/module-info.java:1]: These files must be moved under `src/main/java` as opposed to the current location of `src/main` which places these files outside of all source sets.

**Review by nathanklick — CHANGES_REQUESTED** [review]

**Review by nathanklick — APPROVED** [review]

**Pull request merged**